pkg/apps/strategy/recreate/recreate.go

```diff
@@ -224,32 +214,39 @@ func (s *RecreateDeploymentStrategy) DeployWithAcceptor(from *kapi.ReplicationCo
 	return nil
 }
 
-func (s *RecreateDeploymentStrategy) scaleAndWait(deployment *kapi.ReplicationController, replicas int, retry *kubectl.RetryParams, retryParams *kubectl.RetryParams) (*kapi.ReplicationController, error) {
+func (s *RecreateDeploymentStrategy) scaleAndWait(deployment *kapi.ReplicationController, replicas int, retryTimeout time.Duration) (*kapi.ReplicationController, error) {
 	if int32(replicas) == deployment.Spec.Replicas && int32(replicas) == deployment.Status.Replicas {
 		return deployment, nil
 	}
-	var scaleErr error
-	err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
-		scaleErr = s.scaler.Scale(deployment.Namespace, deployment.Name, uint(replicas), &kubectl.ScalePrecondition{Size: -1, ResourceVersion: ""}, retry, retryParams)
-		if scaleErr == nil {
-			return true, nil
-		}
+	// Update replication controller scale
+	err := wait.PollImmediate(1*time.Second, retryTimeout, func() (bool, error) {
```

Comment:
> @deads2k i want to add watch when @tnozicka upstream PR fixing watch merges, I think we agreed with Tomas that this is just temporary

We don't have watch on subresources.  You really want to avoid using the generic scale client?

Reply:
@deads2k what difference it made? is GetScale() and UpdateScale() going away?

Reply:
actually I might consider the generic scale client... something tells me this:

```
error: couldn't scale router-1 to 1: autoscaling.Scale is not suitable for converting to "v1"
```

might be bugged generated client ?